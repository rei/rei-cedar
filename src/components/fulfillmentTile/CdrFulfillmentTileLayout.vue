<script setup lang="ts">
import { useCssModule, computed } from 'vue';
import mapClasses from '../../utils/mapClasses';
import { Orientation } from '../../types/other';

/** Component for layout of a surface selection */

defineOptions({ name: 'CdrFulfillmentTileLayout' });

interface CdrFulfillmentTileLayoutProps {
  orientation?: Orientation;
}

const props = withDefaults(defineProps<CdrFulfillmentTileLayoutProps>(), {
  orientation: 'horizontal',
});

const style = useCssModule();
const baseClass = 'cdr-fulfillment-tile-layout';

// Manages the props passed along to root
const rootProps = computed(() => {
  const { orientation } = props;
  const classes = [baseClass];

  // Add orientation
  classes.push(`${baseClass}--${orientation}`);

  return { class: mapClasses(style, ...classes) };
});
</script>

<template>
  <div v-bind="rootProps">
    <slot />
  </div>
</template>

<style lang="scss" module src="./styles/CdrFulfillmentTileLayout.module.scss"></style>
