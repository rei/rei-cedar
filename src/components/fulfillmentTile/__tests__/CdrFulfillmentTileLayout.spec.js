import { mount } from '../../../../test/vue-jest-style-workaround.js';
import CdrFulfillmentTileLayout from '../CdrFulfillmentTileLayout.vue';

describe('CdrFulfillmentTileLayout', () => {
  it('has defaults', () => {
    const wrapper = mount(CdrFulfillmentTileLayout, {
      props: {},
      slots: { default: 'X' },
    });
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.classes()).toEqual([
      'cdr-fulfillment-tile-layout',
      'cdr-fulfillment-tile-layout--horizontal',
    ]);
  });

  it('has vertical layout', () => {
    const wrapper = mount(CdrFulfillmentTileLayout, {
      props: { orientation: 'vertical' },
      slots: { default: 'X' },
    });
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.classes()).toEqual([
      'cdr-fulfillment-tile-layout',
      'cdr-fulfillment-tile-layout--vertical',
    ]);
  });
});
