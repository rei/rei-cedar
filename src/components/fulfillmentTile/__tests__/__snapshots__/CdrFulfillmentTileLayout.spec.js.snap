// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CdrFulfillmentTileLayout > has defaults 1`] = `
<div
  class="cdr-fulfillment-tile-layout cdr-fulfillment-tile-layout--horizontal"
>
  <!-- @slot Place elements in the default slot. -->
  
  X
  
</div>
`;

exports[`CdrFulfillmentTileLayout > has vertical layout 1`] = `
<div
  class="cdr-fulfillment-tile-layout cdr-fulfillment-tile-layout--vertical"
>
  <!-- @slot Place elements in the default slot. -->
  
  X
  
</div>
`;
